src/EnlistmentInfo.targets

```diff
@@ -8,10 +8,11 @@
     <DocumentationFile Condition=" !$(IsTestProject) ">$(OutputPath)$(AssemblyName).xml</DocumentationFile>
     <DefineConstants Condition=" '$(TargetFrameworkProfile)' == '' ">$(DefineConstants);DESKTOP</DefineConstants>
   </PropertyGroup>
-
+  
   <ItemGroup>
     <AdditionalFiles Include="$(MSBuildThisFileDirectory)stylecop.json"/>
   </ItemGroup>
 
   <Import Project="NuProj.Extra.targets" Condition=" '$(MSBuildProjectExtension)' == '.nuproj' "/>
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildProjectDirectory), ExportMethods.targets))\ExportMethods.targets" Condition=" $(MSBuildProjectName.Contains('Desktop')) OR $(MSBuildProjectName) == 'BCrypt' "/>
```

Comment:
I'm not sure we need a shared directory for these files at all. I'd like the Kernel32.*exports*.txt file to be in the project directory itself. So we can define that it will be in the desktop project's directory when available, otherwise the portable one. When present, it gets copied to that project's output directory. Then all non-test, non-nuget projects (as you suggested) will produce these .pinvoke.txt files and copy them to their own respective output directories. 
As part of the build also, the project can compare the pinvoke file to the exports file and produce the 3rd file. But if there is no exports file, no comparison file is produced. AppVeyor scoops up all the comparison artifact files, puts them in its flat file listing (as it evidently does I guess) and all we have left is the unsolved problem of how to scrape the artifacts for data to turn into a badge.
How does that sound?